lib/logstash/inputs/beats.rb

```diff
@@ -131,44 +131,40 @@ def register
       LogStash::Logger.setup_log4j(@logger)
     end
 
-    if !@ssl
-      @logger.warn("Beats input: SSL Certificate will not be used") unless @ssl_certificate.nil?
-      @logger.warn("Beats input: SSL Key will not be used") unless @ssl_key.nil?
-    elsif !ssl_configured?
-      raise LogStash::ConfigurationError, "Certificate or Certificate Key not configured"
-    end
+    if @ssl
+      if @ssl_key.nil? || @ssl_key.empty?
+        configuration_error "Missing required setting ssl_key => ... (due ssl => true)"
+      end
+      if @ssl_certificate.nil? || @ssl_certificate.empty?
+        configuration_error "Missing required setting ssl_certificate => ... (due ssl => true)"
+      end
 
-    if @ssl && require_certificate_authorities? && !client_authentification?
-      raise LogStash::ConfigurationError, "Using `verify_mode` set to PEER or FORCE_PEER, requires the configuration of `certificate_authorities`"
-    end
+      if require_certificate_authorities? && !client_authentification?
+        configuration_error "Using verify_mode => '#{@ssl_verify_mode}' requires the configuration of ssl_certificate_authorities => ..."
+      end
 
-    if client_authentication_metadata? && !require_certificate_authorities?
-      raise LogStash::ConfigurationError, "Enabling `peer_metadata` requires using `verify_mode` set to PEER or FORCE_PEER"
+      if client_authentication_metadata? && !require_certificate_authorities?
+        configuration_error "Enabling ssl_peer_metadata => true requires using verify_mode => 'peer' or 'force_peer'"
```

Comment:
Maybe use `ssl_verify_mode` instead of `verify_mode` to be consistent with the configuration name

```suggestion
        configuration_error "Configuring ssl_peer_metadata => true requires `ssl_verify_mode` to be configured with `peer` or `force_peer`"
```